File: 04-04-2024/Sum of all substrings of a number.cpp
//{ Driver Code Starts
#include<bits/stdc++.h>
using namespace std;

// } Driver Code Ends

class Solution
{
    public:
   
    long long sumSubstrings(string s){
        //Your Code Start
        int mod = 1000000007;
        long long ans = 0, temp = 0;
        int n = s.size();
        for (int i = 0; i < n; i++){
            temp = temp * 10 + (i + 1) * (s[i] - '0');
            temp %= mod;
            ans += temp;
            ans %= mod;
        }
        return ans;
        //Your Code End
    }
};


//{ Driver Code Starts.
int main()
{
    //taking total testcases
    int t;
    cin>>t;
    while(t--)
    {
        //taking string
        string s;
        cin>>s;
        Solution ob;
        //calling sumSubstrings() function
        cout << ob.sumSubstrings(s) << endl;
        
    }
    return 0;
}

// } Driver Code Ends

File: 04-04-2024/Sum of all substrings of a number.java
//{ Driver Code Starts
//Initial Template for Java

/*package whatever //do not write package name here */

import java.io.*;
import java.util.*;
class GFG {
	public static void main (String[] args) {
	    
	    //taking input using Scanner class
		Scanner sc=new Scanner(System.in);
		
		//taking total testcases
		int t=sc.nextInt();
		sc.nextLine();
		while(t-->0)
		{
		    //taking the String
		    String s=sc.nextLine();
		    Solution ob = new Solution();
		    //calling method sumSubstrings() 
		    System.out.println(ob.sumSubstrings(s));
		}
	}
}
// } Driver Code Ends




//User function Template for Java

class Solution
{
    //Function to find sum of all possible substrings of the given string.
    //Your Code Start
     private static final long MOD = 1000000007;

    public long sumSubstrings(String s) {
        long ans = 0;
        long temp = 0;
        int n = s.length();
        for (int i = 0; i < n; i++) {
            temp = temp * 10 + (i + 1) * (s.charAt(i) - '0');
            temp %= MOD;
            ans += temp;
            ans %= MOD;
        }
        return ans;
        //Your Code End
    }
}
